src/angular-signature-pad.js

```diff
@@ -60,8 +64,19 @@ function BmSignaturePadController ($scope, $element, $attrs, $window, $log) {
     if (vm.resize) {
       vm.resize({
         $fn: () => {
-          signaturePad.resize(element.clientWidth, element.clientHeight, vm.scaleDown && vm.scaleDown())
-          vm.ngModel.$setViewValue(getSignature())
+          const width = element.clientWidth
+          const height = element.clientHeight
+          // If canvas is empty we dont need to call resize or set the model
+          if (vm.ngModel.$isEmpty()) {
+            canvas.width = width
+            canvas.height = height
+          } else {
+            const contentPreserved = canvasManipulation.resize(canvas, width, height, vm.scaleDown && vm.scaleDown())
```

Comment:
The image degrading will only happen once.

When the phone is turned from landscape to portrait. If the image width does not fit in a portrait view, it will be scaled down. Once the phone is rotated back to landscape, it will not resize the image again as it will still fit in the landscape view. (Or vice versa if the height of the image does not fit in the landscape view).